.pre-commit-hooks.yaml

```
@@ -0,0 +1,24 @@
+- id: poetry-check
+  name: poetry-check
+  description: run poetry check to validate config
+  entry: poetry check
+  language: python
+  language_version: python3
+  pass_filenames: false
+
+- id: poetry-lock
+  name: poetry-lock
+  description: run poetry lock to update lock file
+  entry: poetry lock
+  language: python
+  language_version: python3
+  pass_filenames: false
+
+- id: poetry-export
+  name: poetry-export
+  description: run poetry export to sync lock file with requirements.txt
+  entry: poetry export
+  language: python
+  language_version: python3
+  pass_filenames: false
+  args: ["-f", "requirements.txt", "-o", "requirements.txt"]
```

poetry-export makes a constraints.txt file, not a requirements.txt file

```suggestion
  args: ["-f", "constraints.txt", "-o", "constraints.txt"]
```

---

ah the poetry export is no-longer a valid constraints file https://github.com/python-poetry/poetry/issues/3472#issuecomment-744356551

---

I see more use for/of `requirements.txt` files (correct me if I am wrong here) and these settings are only default which one can locally overwrite if one wants or needs to,